src/crypto/index.js

```diff
@@ -718,17 +726,20 @@ Crypto.prototype.exportRoomKeys = function() {
  * Import a list of room keys previously exported by exportRoomKeys
  *
  * @param {Object[]} keys a list of session export objects
+ * @return {module:client.Promise} a promise which resolves once the keys have been imported
  */
 Crypto.prototype.importRoomKeys = function(keys) {
-    keys.map((session) => {
-        if (!session.room_id || !session.algorithm) {
-            console.warn("ignoring session entry with missing fields", session);
-            return;
-        }
+    return Promise.map(
+        keys, (session) => {
```

Comment:
The naming on this makes me grumpy. Mapping over an array of `keys` *obviously* means mapping over a `session`? Say what? This is either `sessions` and `session` or `keys` and `key`, not `keys` and `session`. Please choose one and stick with it throughout.